pulser-core/pulser/channels.py

```diff
@@ -64,30 +81,101 @@ class Channel(ABC):
         call ``Rydberg.Global(...)``.
     """
 
-    addressing: str
-    max_abs_detuning: float
-    max_amp: float
+    addressing: ADDRESSING
+    max_abs_detuning: Optional[float]
+    max_amp: Optional[float]
     phase_jump_time: int = 0
     min_retarget_interval: Optional[int] = None
     fixed_retarget_t: Optional[int] = None
     max_targets: Optional[int] = None
     clock_period: int = 4  # ns
     min_duration: int = 16  # ns
-    max_duration: int = 67108864  # ns
+    max_duration: Optional[int] = 1000000  # ns
     mod_bandwidth: Optional[float] = None  # MHz
 
     @property
     @abstractmethod
-    def name(self) -> str:
+    def name(self) -> CH_TYPES:
```

Comment:
I agree it would be clearer, but I don't think it is worth it to break backwards-compatibility over. Btw, I changed it in the latest PR to be automatically defined from the class name